File: contest/LeetCode_Weekly_Contest_201/Make_The_String_Great.py
class Solution(object):
    def makeGood(self, s):
        """
        :type s: str
        :rtype: str
        """
        ret = []
        for c in s:
            if len(ret) and \
               ((ret[-1].islower() and c.isupper() and ret[-1] == c.lower()) or
                (ret[-1].isupper() and c.islower() and ret[-1].lower() == c)):
                ret.pop(-1)
            else: ret.append(c)
                
        return "".join(ret)
        

File: contest/LeetCode_Weekly_Contest_201/Make_The_String_Great.js
/**
 * @param {string} s
 * @return {string}
 */
var makeGood = function(s) {
    var ret = [];
    var n = s.length;
    
    for (var i = 0; i < n; ++i) {
        if (ret.length && 
            ('a' <= ret[ret.length-1] && ret[ret.length-1] <= 'z' && 
             'A' <= s[i] && s[i] <= 'Z' && 
             ret[ret.length-1].toLowerCase() === s[i].toLowerCase()) ||
            ('A' <= ret[ret.length-1] && ret[ret.length-1] <= 'Z' && 
             'a' <= s[i] && s[i] <= 'z' && 
             ret[ret.length-1].toLowerCase() === s[i].toLowerCase())) 
            ret.pop();
        else ret.push(s[i]);
    }
    
    return ret.join('');
};
